### PR TITLE
Marketplace: Add test for plugin author page

### LINKED ISSUE #66532: Marketplace: Add test to check plugin author page is correctly linked
To prevent #66520 from recurring, we would like an ~~e2e~~ component test to check this flow:

- ~~Visit https://wordpress.com/plugins/wordpress-seo-premium~~
- ~~Click on the plugin developer~~
- ~~You should be redirected to /plugins/?s=developer:"toast"~~
- Render `PluginDetailsHeader`
- Check for the correct author URL in href attribute

### PR DESCRIPTION
**Pull request by candy02058912**
#### Proposed Changes

* Add test to check plugin author page is correctly linked to prevent bugs like #66520.

#### Testing Instructions

* ~~Check out how to run e2e tests [here](https://github.com/Automattic/wp-calypso/tree/trunk/test/e2e#quick-start)~~
* ~~`yarn jest <path-to-calypso>/test/e2e/specs/plugins/plugins__developer.ts`~~
* `yarn test-client  -- <path-to-calypso>/client/my-sites/plugins/plugin-details-header`

#### Pre-merge Checklist

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Fixes #66532

### CODE REVIEW

**Review by lupus2k — APPROVED** [review]
LGTM. Good Job @candy02058912 🎉

File: test/e2e/specs/plugins/plugins__search-developer.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+	it( 'Visit Yoast plugin page', async function () {
+		pluginsPage = new PluginsPage( page );
+		await pluginsPage.visitPage( 'wordpress-seo-premium' );
+	} );
+
+	it( 'Developer shows up in search term', async function () {
+		await pluginsPage.clickDeveloperLink();
+		await pluginsPage.validateExpectedSearchResultFound( 'Yoast SEO Premium' );
+		await pluginsPage.validateSearchInputValue( 'developer:"yoast"' );
```
Comment (worldomonation) [inline test/e2e/specs/plugins/plugins__search-developer.ts]: **comment**: while this test is good, ideally this is written as a React component test that does not rely on the Calypso environment for testing (unless it is required due to some reason).

Comment (candy02058912) [reply test/e2e/specs/plugins/plugins__search-developer.ts]: Thanks for bringing this up. I decided to write the component test and the [issue](https://github.com/Automattic/wp-calypso/issues/66532) can still be addressed.

**Review by lupus2k — APPROVED** [review]
LGTM! Good job 🚀

File: test/e2e/specs/plugins/plugins__search-developer.ts (hunk 1 above)
Comment (lsl) [reply test/e2e/specs/plugins/plugins__search-developer.ts]: Sorry @candy02058912, looks like I shouldn't have suggested an e2e test for this!

**Pull request merged**